Remove view box

bgschiller opened the pull request:
fixes #100, #174, and #132.

This is the longer-term fix whose arrival was foretold in #184 🔮  👻 .

The main point of this PR is removing this line:
```
this.$refs.grid.setAttribute('viewBox', `0 0 ${this.max_x - this.min_x} ${this.max_y - this.min_y}`);
```
Almost everything else was removing the workarounds that were added to accommodate that line, or adding things to correct for the fact that we're now working on the scale of pixels instead of real-world-units.

That line said "scale everything inside the svg so that when I say "1" I mean "1 meter" or "1 foot", instead of "1 pixel" as would be the default. That meant:

- font sizes now needed to be scaled. "12" wasn't "12pt", it was "12m", so they were scaled down to avoid taking up the whole screen. This didn't always work, as in #100.
- When working in a small region, say, `max_x - min_x` less than 20 or so, there was a noticeable offset between where the grid was drawn, and where click events registered. This is because d3 applies a 0.5 offset when drawing axes (https://groups.google.com/forum/#!topic/d3-js/-Sw0Wdnkeko) in order to fix a rendering bug in safari. 0.5 was intended to mean '0.5pixels', but was being interpreted as '0.5 ft'.

Also in this PR, I changed the grid and polygon drawing code to update existing DOM nodes if possible, rather than destroying and creating new ones. Not sure if that actually improves things, since vue was going to do a virtual dom diff anyway. ¯\\\_(ツ)_/¯

bgschiller committed:
Merge branch 'scale-work' into remove-view-box

bgschiller committed:
scales are working

bgschiller committed:
scale work with zoom

bgschiller committed:
adjust font size

bgschiller committed:
remove most scaleXs

bgschiller committed:
preserve aspect ratio

bgschiller committed:
remove call to center grid

bgschiller committed:
re render grid after zoom (who knows why, this stuff is magic)

bgschiller committed:
update grid, rather than re-draw

bgschiller committed:
fix disappearing space text

bgschiller committed:
redo zoom scales on import

bgschiller committed:
Merge branch 'develop' into remove-view-box

gh-23378 commented on src/store/modules/project/index.js:
```diff
@@ -13,14 +13,14 @@ export default {
     grid: {
       visible: true,
-      spacing: 50
+      spacing: 50,
```
let's decrease the default spacing and update our y values for when the grid loads

gh-23378 commented on src/components/Grid/Grid.vue:
```diff
@@ -53,6 +52,7 @@ export default {
 
     // add event listeners
     this.$refs.grid.addEventListener('reloadGrid', this.renderGrid);
+    this.$refs.grid.addEventListener('reloadGridScales', this.reloadGridAndScales);
```
when you zoom too far out, disable the grid for performance

gh-23378 commented on src/components/Grid/Grid.vue:
```diff
@@ -223,17 +225,17 @@ export default {
     transform(newTransform, lastTransform) {
       // hide grid if zoomed out enough
-      this.forceGridHide = (newTransform.kAbs < 0.1);
+      this.forceGridHide = (newTransform.k < 0.1);
```
oops, didn't see this. looks like we're already disabling the grid but there's something stopping it from happening

gh-23378 commented on src/store/utilities/importFloorplan.js:
```diff
@@ -79,5 +79,5 @@ export default function importFloorplan(context, payload) {
   });
 
-  document.getElementById('svg-grid').dispatchEvent(new Event('reloadGrid'));
+  document.getElementById('svg-grid').dispatchEvent(new Event('reloadGridScales'));
```
having a strange effect on image imports

bgschiller committed:
use the new selection.merge instead of a full re-render

just happened to read https://medium.com/@mbostock/what-makes-software-good-943557f8a488
which had exactly the thing I needed to see to fix that bug

bgschiller committed:
lots of work on grid rendering

bgschiller committed:
limit zoom in

bgschiller committed:
no need to load a second time

bgschiller committed:
Merge branch 'image-refactor' into remove-view-box

bgschiller committed:
Fix original resolution

bgschiller committed:
fix image positioning

bgschiller committed:
images scaling, rendering correctly, aligned with grid

bgschiller committed:
Fix bug where image control points would grow when zooming in

bgschiller committed:
Remove debug circles

bgschiller committed:
remove more debug code

bgschiller commented:
@devKate Images are working very well now. Lots of changes, so it's probably worth another review.

gh-23378 commented:
Nice! Looks good :)
